pkg/prowgen/prowgen.go

```diff
@@ -275,6 +195,7 @@ func generatePeriodicForTest(name string, info *ProwgenInfo, podSpec *corev1.Pod
 		BaseRef: info.Branch,
 	}
 	if pathAlias != nil {
+		base.PathAlias = ""
```

Comment:
Yeah this was a surprising one, this is actually how periodics have different `JobBase`s than other jobs, when `PathAlias` is present. When I do not clean this up, we generate jobs in o/release differently. I am not exactly sure why it is that way though.

Previously, this was happening because `generatePeriodicForTest` did not pass its `pathAlias` when generating the `JobBase`:
https://github.com/openshift/ci-tools/blob/cee280168ade9363d13d27ad768bc3e5d7c06299/pkg/prowgen/prowgen.go#L267-L268

/hold
I think I can improve this, I agree this is a little hacky and surprising.